Doc/whatsnew/3.13.rst

```diff
@@ -484,45 +484,139 @@ Optimizations
+  loads, speeding them up and also allowing for more constant propagation.
 
-* When the ``--enable-experimental-jit`` option is used, the optimized
-  Tier 2 IR is translated to machine code, which is then executed.
-  This does not require additional runtime options.
+* This section is non-exhaustive and will be updated with further
+  optimizations, until CPython 3.13's beta release.
 
-* The machine code translation process uses an architecture called
-  *copy-and-patch*. It has no runtime dependencies, but there is a new
-  build-time dependency on LLVM.
+(Tier 2 optimizer contributed by Ken Jin, with implementation help
+by Guido van Rossum, Mark Shannon, and Jules Poon. Special thanks
+to Manuel Rigger and Martin Henz.)
+
+
+Execution Engine
+----------------
+
+There are two execution engines for Tier 2 IR:
+the Tier 2 interpreter and the Just-in-Time (JIT) compiler.
+
+The Tier 2 interpreter is mostly intended for debugging
+the earlier stages of the optimization pipeline. If the JIT is not
+enabled, the Tier 2 interpreter can be invoked by passing Python the
+``-X uops`` option or by setting the ``PYTHON_UOPS`` environment
+variable to ``1``.
+
+The second is the JIT compiler. When the ``--enable-experimental-jit``
+build-time option is used, the optimized Tier 2 IR is translated to machine
+code, which is then executed. This does not require additional
+runtime options.
+
+The machine code translation process uses a technique called
+*copy-and-patch*. It has no runtime dependencies, but there is a new
+build-time dependency on `LLVM <https://llvm.org>`_.
+The main benefit of this technique is
+fast compilation, reported as orders of magnitudes faster versus
+traditional compilation techniques in the paper linked below. The code
+produced is slightly less optimized, but suitable for a baseline JIT
+compiler.
+
+(Copy-and-patch JIT compiler contributed by Brandt Bucher,
+directly inspired by the paper
+`Copy-and-Patch Compilation <https://fredrikbk.com/publications/copy-and-patch.pdf>`_
+by Haoran Xu and Fredrik Kjolstad. For more information,
+`a talk <https://youtu.be/HxSHIpEQRjs?si=RwC78FcXrThIgFmY>`_ is available.)
+
+
+Results and Future Work
+-----------------------
+
+The final performance results will be updated before CPython 3.13's beta release.
+
+The JIT compiler is rather unoptimized, and serves as the foundation
+for significant optimizations in future releases. As such, we do not
+expect the first iteration of the JIT compiler to produce a significant
+speedup.
+
+
+About
+-----
 
-(JIT by Brandt Bucher, inspired by a paper by Haoran Xu and Fredrik Kjolstad.
-Tier 2 IR by Mark Shannon and Guido van Rossum.
-Tier 2 optimizer by Ken Jin.)
+This work was done by the Faster CPython team, and many other external
+contributors. The team consists of engineers from Microsoft, Meta,
+Quansight, and Bloomberg, who are either paid in part to do this, or
+volunteer in their free time.
```

Comment:
I don't see the point of this section -- contributors are already listed above. If and when Microsoft wants its name in the credits I will let you know, but not now.

Reply:
> We had a similar credits section in 3.11, so this is mirroring that, but I will remove it. https://docs.python.org/3/whatsnew/3.11.html#about

Thanks. It's tradition to credit individuals regardless of who paid for their time.